drivers: ams-tmd2635: add driver code for proximitory sensor tmd2635

yunheQin opened the pull request:
This PR adds basic support of proximitory sensor tmd2635. The sensor was tested with MIMXRT1062xxx6A for monitoring position value of sample with following output.

![Untitled picture](https://github.com/zephyrproject-rtos/zephyr/assets/137849449/bfb0f573-1f59-4456-b20b-a21902524861)

Commit:
feat: samples: extend shell example

	extend the shell example to test:
		ADC
		CAN
		I2C
		NET
		PWM
		GPIO

Signed-off-by: Florian Guhl <florian.guhl@zeiss.com>
Signed-off-by: Dipak Shetty <dipak.shetty@zeiss.com>

Commit:
fix: sensors: patch ism330dhcx_spi.c

Signed-off-by: Florian Guhl <florian.guhl@zeiss.com>

Commit:
feat: samples: subsys: shell: shell_module: extend shell example

Extended to test TRINAMIC_TMC5041 module

Signed-off-by: Dipak Shetty <dipak.shetty@zeiss.com>

Commit:
feat: #550460: add ism330dhcx shell

Signed-off-by: Florian Guhl <florian.guhl@zeiss.com>

jilaypandya committed:
feat:add adc_shell to prj.conf

Commit:
feat: activate can fd mode in prj.conf

Signed-off-by: Dipak Shetty <dipak.shetty@zeiss.com>

martinjaeger commented on drivers/can/can_shell.c:
This change seems unrelated to the title of this PR. Please split into separate PR.

martinjaeger commented on drivers/sensor/tmd2635/tmd2635.c:
```diff
@@ -0,0 +1,514 @@
+
+	if (chan == SENSOR_CHAN_PROX) {
+		/* inverting sensor data to fit Zephyr */
+		// LOG_INF("original data: %X ..",data->pdata);
```
Use C90 style comments only and no dead (commented-out) code, please.

martinjaeger commented on drivers/sensor/tmd2635/tmd2635.c:
```diff
@@ -0,0 +1,514 @@
+		LOG_ERR("Failed reading calibra status");
+		return ret;
+	}
+	// else{
+	// 		if (calibra_status & TMD2635_CALIBSTAT_CALIB_FINISHED )
+	// 		{
+	// 			status
+	// 		}
+
+	// }
```
No dead code. Please remove.

martinjaeger commented on samples/subsys/shell/shell_module/prj.conf:
Unrelated changes for this PR.

martinjaeger reviewed (changes requested):
Dear @yunheQin,

thanks for your contribution.

The changes to the shell seem to be unrelated to this PR and should be split into a dedicated PR.

Please have a look at the link below for commit message guidelines. The are almost correct, but we don't use feat/fix/... prefixes. 
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines

See the additional comments for some first feedback.

yunheQin committed:
Revert "feat: aopyright (c) 2023 Phytec Messtechnik GmbH"

This reverts commit b4a42397bfcc50982105022f627cda0326ccd5a9.
this commits reverts last push

Commit:
feat: aopyright (c) 2023 Phytec Messtechnik GmbH

Signed-off-by: Dipak Shetty <dipak.shetty@zeiss.com>
Signed-off-by: Yunhe Qin <yun-he.qin@zeiss.com>

yunheQin committed:
drivers: sensor: ams-tmd2635: add initial driver files

This commit added initial driver files of ams-tmd2635;

This commit modified the KConfig and CMakeLists.txt in the
"drivers/sensor" folder to add compatiable with ams-tmd2635;

This commit add the binding file to use the ams-tmd2635.

Signed-off-by: Yunhe Qin <yun-he.qin@zeiss.com>

henrikbrixandersen reviewed (changes requested):
-1 on the CAN shell changes in this PR.

FlorianGuhl-zeiss commented:
@yunheQin can you please remove @dipakgmx and my commits from this PR?

martinjaeger commented:
@yunheQin Please don't open a new PR (if that is your plan), but fix this one instead. If you need help with git, just let us know.

yunheQin commented:
All issues above have been solved, so please Reopen this PR.

yunheQin commented:
The above all issues have been solved, and this PR should be opened to finish the code update.

henrikbrixandersen commented:
> The above all issues have been solved, and this PR should be opened to finish the code update.

This PR cannot be reopened since the branch was either force-pushed or recreated while the PR was closed.

martinjaeger commented:
Well, that's unfortunate, but in that case you'll have to open a new PR @yunheQin.

I had a look into your branch and there is still a merge commit in it: https://github.com/yunheQin/zephyr/commits/feature/add-proximitory-sensor-tmd2635

Please remove that before opening the new PR. As stated in the docs linked above, merge commits are not accepted.
